Clean up network packet handling

valarnin opened the pull request:
This PR has now been tested (see https://github.com/OverlayPlugin/OverlayPlugin/pull/345#issuecomment-2134314178) and is ready for a full review and merge.

The goals of this PR are as follows:

1. Remove boilerplate where possible
2. Ensure that all network packet handling accounts for potential region differences
3. Implement more robust error logging

valarnin commented:
@wexxlee I think this is ready for some initial thoughts. Maybe @xpdota as well, since you've written some stuff for network packet handling.

xpdota commented:
Looks like some good improvements to me.

wexxlee commented on OverlayPlugin.Core/NetworkProcessors/LineActorControlSelfExtra.cs:
```diff
@@ -38,146 +36,87 @@ public class LineActorControlSelfExtra
+
+                if (!AllowedActorControlCategories.Contains(category)) return null;
+
+                var param1 = Get<UInt32>("param1");
+                var param2 = Get<UInt32>("param1");
+                var param3 = Get<UInt32>("param1");
+                var param4 = Get<UInt32>("param1");
+                var param5 = Get<UInt32>("param1");
+                var param6 = Get<UInt32>("param1");
```
Copy-paste miss on the param names?  i.e., shouldn't it be:
```var param2 = Get<UInt32>("param2");``` etc.

wexxlee reviewed (commented):
LGTM as well.  Haven't done side-by-side testing to confirm (and I don't have a kr/cn client to test against), but approach-wise, this seems like a nice bit of cleanup.  Just one comment on `ActorControlSelfExtra`.

valarnin commented:
Alright, this is technically ready for general review. I still need to do a bunch of testing, but the overall refactoring is complete.

valarnin commented:
I have tested:

- LineAbilityExtra
  - Fixed a few issues with memory and pointers, as well as struct offsets
- LineActorCastExtra, no issues
- LineActorControlExtra & LineActorControlSelfExtra
  - For some reason `string.Format` was throwing an exception due to `:X8` and `:X4`. No clue why, it didn't do that before the refactor, I blame Microsoft. Changed it to just use `$"{...:X8}"` instead.
- LineActorMove, no issues
- LineBattleTalk2
- LineNpcYell
- LineCountdown
- LineCountdownCancel

Still needs tested:
- LineActorSetPos
- LineCEDirector
- LineFateControl
- LineMapEffect
- LineRSV (we can't really test this currently, since there aren't any RSVs left in the game until probably the first savages of Dawntrail)
- LineSpawnNPCExtra

wexxlee commented:
Next couple of days are pretty crazy for me, but I should have some time Fri/Sat to give this a run.

wexxlee reviewed (approved):
Gave this a workout, and looks good.  This is a nice bit of cleanup & bug fixes.
